File: src/main/java/com/github/sorusclient/client/adapter/IPotionEffect.java
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.adapter;

public interface IPotionEffect {
    String getDuration();
    String getName();
    int getAmplifier();
    PotionType getType();
}

File: src/main/kotlin/com/github/sorusclient/client/adapter/IPotionEffect.kt
package com.github.sorusclient.client.adapter

interface IPotionEffect {
    val duration: String
    val name: String?
    val amplifier: Int
    val type: PotionType?

    enum class PotionType {
        SPEED, SLOWNESS, HASTE, MINING_FATIGUE, STRENGTH, INSTANT_HEALTH, INSTANT_DAMAGE, JUMP_BOOST, NAUSEA, REGENERATION, RESISTANCE, FIRE_RESISTANCE, WATER_BREATHING, INVISIBILITY, BLINDNESS, NIGHT_VISION, HUNGER, WEAKNESS, POISON, WITHER, HEALTH_BOOST, ABSORPTION, SATURATION, UNKNOWN
    }
}